core/controller/cluster/controller_basic_v2.go

```diff
@@ -580,3 +581,26 @@ func (c *controller) customizeCreateReqBuildTemplateInfo(ctx context.Context, r
 	}
 	return buildTemplateInfo, nil
 }
+
+func (c *controller) Like(ctx context.Context, clusterID uint, like *WhetherLike) (err error) {
+	// get current user
+	currentUser, err := common.UserFromContext(ctx)
+	if err != nil {
+		return perror.WithMessage(err, "no user in context")
+	}
+
+	if like != nil {
+		if like.IsFavorite {
+			collection := collectionmodels.Collection{
+				ResourceID:   clusterID,
+				ResourceType: common.ResourceCluster,
+				UserID:       currentUser.GetID(),
+			}
+			_, err := c.collectionManager.Create(ctx, &collection)
```

Comment:
Is it possible that more than one collection record could be created with the same cluster if user passes the same clusterID and IsFavorite=true in two different requests?

Reply:
we got unique key in database